numba/targets/arrayobj.py

```diff
@@ -1480,6 +1487,12 @@ def array_transpose_vararg(context, builder, sig, args):
     new_sig, new_args = vararg_to_tuple(context, builder, sig, args)
     return array_transpose_tuple(context, builder, new_sig, new_args)
 
+@overload(np.transpose)
```

Comment:
It does.

Questions about tests for this: what do I do with test_array_method tests? That patch only has transpose with axes in that file. Do you reckon I should move the basic functionality tests to test_array_manipulation?

Reply:
I think the issue here is that `.T/.transpose()` is a method and `np.transpose()` is a NumPy function and that the gluing together of the testing of these would break the traditional separation found in the Numba test regime? I think in this case it's fine to put them both in the same place, and in the case of `transpose` as most uses I've seen are the `ndarray` method then the tests for `np.transpose` ought to go where the tests for that is. You can always import `from_generic` into the `test_array_methods` if that helps?

Reply:
I think moving the axes methods into ````test_array_manipulation```` is the better way to go, honestly. The tests for ````transpose```` in ````test_array_methods```` are very limited, can't see a clear theme here.

Next up I'm planning to do some refactoring of tests, so I'll probably have more questions then.